client/src/controllers/AutoFieldController.ts

```diff
@@ -24,6 +24,12 @@ export class AutoFieldController extends Controller<
       );
     }
 
+    // Update form action based on selected option
+    if (this.element.type === 'select-one') {
+      const selectedOption = this.element.options[this.element.selectedIndex];
+      form.action = selectedOption.dataset.action;
+    }
+
```

Comment:
@Aadijindal07  it looks like have got some changes from some other PRs here.

You may need to create a separate branch for each of the PRs you want to put together.

This article has some good guidelines and tips for making your first contributions - https://wagtail.org/blog/ten-tasty-ingredients-for-a-delicious-pull-request/